### ISSUE TITLE
[UI-Shell] Tab navigation of Navbar includes hidden elements

### ISSUE THREAD

**Issue by eddybrando** [Qiskit/web-components#152]
### Steps to reproduce the problem

1. Open the Navbar on a desktop-sized viewport
2. Use the tab to navigate thought the Navbar links and see the target links
3. Move to the next element after "Documentation" with the tab key
4. See how hidden links from the mobile menu are being selected (like overview)

### What is the current behavior?

Hidden links are accessible via tab.

### What is the expected behavior?

Hidden links (like the links from the mobile menu when on a desktop-sized viewport) should not be accessible via tabs.

### Screenshots

In this screen recording, I change the link to the overview page that should only be accessible in the mobile menu.

Then I use tabs to navigate the Navbar.

Please watch the target links at the bottom of the window when I select a new link from the Navbar, and see how after I select Dropdown, the overview link from the mobile menu is selected, even thought the mobile menu should be invisible / inaccessible in this viewport.

https://user-images.githubusercontent.com/22047320/190395437-8da400e8-43ed-4a10-993e-2e2d5ceb6912.mov

### Device

MacBook Pro

### OS

MacOS

### Browser

Chrome
